### ISSUE TITLE
Power9 ppc64le will not parse JSON

### ISSUE THREAD

**Issue by markplagge** [Tencent/rapidjson#1378]
I've got rapidjson embedded in an application currently, and I'm trying to get it to run on an IBM Power9 ppc64le system. Everything works fine when compiled on an Intel 64 bit system using both icc and GCC 8.1.0. However, I've run into issues on the ppc64le system. 

Specifically, when trying to parse a JSON document, I get the following
(Example Code)
`Document jx;
jx.ParseInsitu<kParseCommentsFlag>(buffer);`

Will give a parsing result 4: "Missing a name for object member".

Any attempt to access the members of `jx` using  the format:
`jx["model"].GetString();`
for example, will throw:
`Assertation IsObject() failed`

I want to stress that this code (including the added test code) works fine on the Intel system. 

Initially, the program opened a file and used the file parsing features of RapidJson directly on the file, which produced the error. To try and isolate the bug, I've done the following:

1. Created a static const string - and fed that into a normal parser (js.Parse)
2. Created a dynamically allocated C string (char buffer*), and filled it with some dummy JSON data.
3. Same as (2), but tried the ParseInsitu method.
4. Disabled -mcpu / -qarch options
5. Made sure Cx11 was enabled and working
6. Checked that 48 Bit Intel Packing was disabled

On this platform, the Power9 is running in little endian mode. I've tried compiling with both GCC and the IBM XL C compilers, and get the same result. To ensure that RapidJson is in little-endian mode, I've printed out the state of `RAPIDJSON_ENDIAN`, and it returns 0. 

Is there some incompatibility going on? Does anyone have any idea why this would happen? 

Thanks!

**Comment by Bu11etmagnet** [comment]
Obviously, if ParseInsitu failed, any attempt to access the document will fail.

Unfortunately, my psychic powers are insufficient to determine the content of `buffer`. Without that information, I have no idea why the parsing failed.

**Comment by miloyip** [comment]
I suspect that the parsing of comment is not working thus reporting that error message.
And why doesn't it work? Another suspect is that the program has more than one version of RapidJSON and the linker selected an older version which does not support comment parsing.
Can you try to make a minimal, reproducible test to verify that?
